src/api/app/models/project.rb

```diff
@@ -1404,13 +1404,10 @@ def bsrequest_repos_map(project)
 
   def self.valid_name?(name)
     return false unless name.kind_of? String
-    # this length check is duplicated but useful for other uses for this function
-    return false if name.length > 200 || name.blank?
     return false if name == "0";
-    return false if name =~ %r{^[_\.]}
-    return false if name =~ %r{::}
+    return false if name =~ /::/
     return false if name.end_with?(':')
-    return true if name =~ /\A\w[-+\w\.:]*\z/
+    return true if name =~ /\A[a-zA-Z0-9][-+\w\.:]{0,199}\z/
```

Comment:
but as it was said in the comment the length checked is already duplicated, so not that important to make it explicit. And if you want to know what the function does you have to find out what the regular expression does so I think that there is not much difference.
If it is longer it may be faster, but if not I would say it the same as you have to check all the character in the String anyway.